lib/trino-hive-formats/src/main/java/io/trino/hive/formats/DistinctMapKeys.java

```diff
@@ -66,7 +66,7 @@ public boolean[] selectDistinctKeys(Block keyBlock)
             if (keyBlock.isNull(i)) {
                 continue;
             }
-            int hash = getHashPosition(keyBlock, i, hashTableSize);
+            int hash = getHashPosition(keyBlock, i, hashTable.length);
```

Comment:
It looks like `hashTableSize` is now unused, we should probably delete the local variable as well since it's confusing and brings in logic that doesn't apply to this this hash table implementation like `HASH_MULTIPLIER`.

Reply:
Actually, on second thought it seems like the right behavior is to initialize `hashTableBuffer` to be at least `hashTableSize` and not `bufferSize` since otherwise hash collisions will be very common and costly. cc @dain